File: packages/python/examples/basic_usage.py
"""
Example usage of the Muninn SDK.

This file demonstrates various features of the Muninn Python SDK.
Run with: python examples/basic_usage.py
"""

from muninn import MuninnClient
from muninn.exceptions import MuninnError
from muninn.types import MemoryType, Visibility


def main():
    # Initialize the client with your API key
    # Get your API key from: https://muninn-supabase.vercel.app
    API_KEY = "muninn_live_your_key_here"

    # Create client
    client = MuninnClient(api_key=API_KEY)

    # ========================================
    # Example 1: Store a memory
    # ========================================
    print("=" * 50)
    print("Example 1: Storing a memory")
    print("=" * 50)

    memory = client.memories.store(
        content="User prefers dark mode in the application settings",
        type=MemoryType.SEMANTIC,
        entities=["user_123"],
        metadata={"category": "ui_preference", "source": "settings_page"},
    )

    print(f"✓ Stored memory with ID: {memory.id}")
    print(f"  Content: {memory.content}")
    print(f"  Type: {memory.type}")
    print(f"  Embedding generated: {memory.embedding_generated}")
    print()

    # Store another memory
    memory2 = client.memories.store(
        content="User frequently asks about pricing plans",
        type=MemoryType.EPISODIC,
        entities=["user_123", "pricing"],
        metadata={"frequency": "high"},
    )

    print(f"✓ Stored another memory with ID: {memory2.id}")
    print()

    # ========================================
    # Example 2: Search memories
    # ========================================
    print("=" * 50)
    print("Example 2: Searching memories")
    print("=" * 50)

    results = client.memories.search(
        query="user interface preferences",
        limit=10,
        threshold=0.3,
    )

    print(f"Found {results.count} results (search type: {results.search_type})")
    for mem in results.results:
        print(f"  - [{mem.type.value}] {mem.content[:50]}...")
    print()

    # ========================================
    # Example 3: Get single memory
    # ========================================
    print("=" * 50)
    print("Example 3: Getting a single memory")
    print("=" * 50)

    fetched = client.memories.get(memory.id)
    print(f"✓ Fetched memory: {fetched.id}")
    print(f"  Content: {fetched.content}")
    print(f"  Metadata: {fetched.metadata}")
    print(f"  Created: {fetched.created_at}")
    print()

    # ========================================
    # Example 4: Filter by type
    # ========================================
    print("=" * 50)
    print("Example 4: Filtering by memory type")
    print("=" * 50)

    episodic_results = client.memories.search(
        query="user interaction",
        type=MemoryType.EPISODIC,
        limit=5,
    )

    print(f"Found {episodic_results.count} episodic memories")
    print()

    # ========================================
    # Example 5: Delete a memory
    # ========================================
    print("=" * 50)
    print("Example 5: Deleting a memory")
    print("=" * 50)

    deleted = client.memories.delete(memory2.id)
    print(f"✓ Deleted memory {memory2.id}: {deleted}")
    print()

    # ========================================
    # Example 6: Health check
    # ========================================
    print("=" * 50)
    print("Example 6: Health check")
    print("=" * 50)

    health = client.health()
    print(f"Status: {health.get('status')}")
    print(f"Service: {health.get('service')}")
    print(f"Version: {health.get('version')}")
    print()

    # Clean up
    client.close()
    print("✓ All examples completed successfully!")


if __name__ == "__main__":
    main()

File: packages/typescript/examples/basic-usage.ts
/**
 * Example usage of the Muninn SDK.
 *
 * This file demonstrates various features of the Muninn TypeScript SDK.
 * Run with: npx tsx examples/basic-usage.ts
 */

import { MuninnClient, MuninnError } from '../src';
import type { MemoryType } from '../src';

async function main() {
  // Initialize the client with your API key
  // Get your API key from: https://muninn-supabase.vercel.app
  const API_KEY = 'muninn_live_your_key_here';

  // Create client
  const client = new MuninnClient({ apiKey: API_KEY });

  try {
    // ========================================
    // Example 1: Store a memory
    // ========================================
    console.log('='.repeat(50));
    console.log('Example 1: Storing a memory');
    console.log('='.repeat(50));

    const memory = await client.memories.store({
      content: 'User prefers dark mode in the application settings',
      type: 'semantic' as MemoryType,
      entities: ['user_123'],
      metadata: { category: 'ui_preference', source: 'settings_page' },
    });

    console.log(`✓ Stored memory with ID: ${memory.id}`);
    console.log(`  Content: ${memory.content}`);
    console.log(`  Type: ${memory.type}`);
    console.log(`  Embedding generated: ${memory.embedding_generated}`);
    console.log();

    // Store another memory
    const memory2 = await client.memories.store({
      content: 'User frequently asks about pricing plans',
      type: 'episodic' as MemoryType,
      entities: ['user_123', 'pricing'],
      metadata: { frequency: 'high' },
    });

    console.log(`✓ Stored another memory with ID: ${memory2.id}`);
    console.log();

    // ========================================
    // Example 2: Search memories
    // ========================================
    console.log('='.repeat(50));
    console.log('Example 2: Searching memories');
    console.log('='.repeat(50));

    const results = await client.memories.search({
      query: 'user interface preferences',
      limit: 10,
      threshold: 0.3,
    });

    console.log(`Found ${results.count} results (search type: ${results.search_type})`);
    for (const mem of results.results) {
      console.log(`  - [${mem.type}] ${mem.content.slice(0, 50)}...`);
    }
    console.log();

    // ========================================
    // Example 3: Get single memory
    // ========================================
    console.log('='.repeat(50));
    console.log('Example 3: Getting a single memory');
    console.log('='.repeat(50));

    const fetched = await client.memories.get(memory.id);
    console.log(`✓ Fetched memory: ${fetched.id}`);
    console.log(`  Content: ${fetched.content}`);
    console.log(`  Metadata: ${JSON.stringify(fetched.metadata)}`);
    console.log(`  Created: ${fetched.created_at}`);
    console.log();

    // ========================================
    // Example 4: Filter by type
    // ========================================
    console.log('='.repeat(50));
    console.log('Example 4: Filtering by memory type');
    console.log('='.repeat(50));

    const episodicResults = await client.memories.search({
      query: 'user interaction',
      type: 'episodic',
      limit: 5,
    });

    console.log(`Found ${episodicResults.count} episodic memories`);
    console.log();

    // ========================================
    // Example 5: Delete a memory
    // ========================================
    console.log('='.repeat(50));
    console.log('Example 5: Deleting a memory');
    console.log('='.repeat(50));

    const deleted = await client.memories.delete(memory2.id);
    console.log(`✓ Deleted memory ${memory2.id}: ${deleted}`);
    console.log();

    // ========================================
    // Example 6: Health check
    // ========================================
    console.log('='.repeat(50));
    console.log('Example 6: Health check');
    console.log('='.repeat(50));

    const health = await client.health();
    console.log(`Status: ${health.status}`);
    console.log(`Service: ${health.service}`);
    console.log(`Version: ${health.version}`);
    console.log();

    console.log('✓ All examples completed successfully!');
  } catch (error) {
    if (error instanceof MuninnError) {
      console.error(`Muninn Error [${error.statusCode}]: ${error.message}`);
    } else {
      console.error('Unexpected error:', error);
    }
  }
}

main();